packages/logger/src/index.ts

```diff
@@ -93,6 +96,100 @@ function createDisabledLogger (namespace: string): debug.Debugger {
   return logger
 }
 
+export interface ComponentLoggerOptions {
+  prefixLength: number
+  suffixLength: number
+}
+
+/**
+ * Create a component logger that will prefix any log messages with a truncated
+ * peer id.
+ *
+ * @example
+ *
+ * ```js
+ * import { peerLogger } from '@libp2p/logger'
+ * import { peerIdFromString } from '@libp2p/peer-id'
+ *
+ * const peerId = peerIdFromString('12D3FooBar')
+ * const logger = peerLogger(peerId)
+ *
+ * const log = logger.forComponent('my-component')
+ * log.info('hello world')
+ * // logs "12D3…oBar:my-component hello world"
+ * ```
+ */
+export function peerLogger (peerIdOrString: PeerId, options: Partial<ComponentLoggerOptions> = {}): ComponentLogger {
```

Comment:
Let's add some tests for these

Reply:
I've added a test for the peer id truncation, not sure what else there is to test here.